src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/TempMultipleAlignmentsReclassifier.java

```diff
@@ -0,0 +1,211 @@
+                                      lowUniquenessMappings = new ArrayList<>(originalConfiguration.size());
+
+        for (final AlignmentInterval alignment : originalConfiguration) {
+            if (alignment.mapQual >= mapQThresholdInclusive)
+                selectedAlignments.add(alignment);
+            else
+                lowUniquenessMappings.add(alignment);
+        }
+
+        // second pass, the slower one, is to remove alignments offering low READ UNIQUENESS,
+        // i.e. with only a very short part of the read being uniquely explained by this particular alignment;
+        // the steps are:
+        //      search bi-directionally until cannot find overlap any more, subtract from it all overlaps.
+        //      This gives unique read region it explains.
+        //      If this unique read region is "short": shorter than {@code uniqReadLenInclusive}), drop it.
+
+        // each alignment has an entry of a tuple2, one for max overlap front, one for max overlap rear,
+        // max overlap front is a tuple2 registering the index and overlap bases count
+        final List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> maxOverlapMap =
+                getMaxOverlapPairs(selectedAlignments);
+
+        final List<Integer> idxToRemove = new ArrayList<>(selectedAlignments.size());
+        for (int i = 0; i < selectedAlignments.size(); ++i) {
+            final AlignmentInterval cur = selectedAlignments.get(i);
+            final Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> maxOverlapFrontAndRear = maxOverlapMap.get(i);
+            final int maxOverlapFront = Math.max(0, maxOverlapFrontAndRear._1._2);
+            final int maxOverlapRead = Math.max(0, maxOverlapFrontAndRear._2._2);
+
+            final int uniqReadSpan = cur.endInAssembledContig - cur.startInAssembledContig + 1 - maxOverlapFront - maxOverlapRead;
+            if (uniqReadSpan < uniqReadLenInclusive)
+                idxToRemove.add(i);
+        }
+
+        if ( idxToRemove.isEmpty() )
+            return new Tuple2<>(selectedAlignments, lowUniquenessMappings);
+
+        // removing in reverse order so that iterators are not invalidated if we were to remove from start
+        final ListIterator<Integer> rit = idxToRemove.listIterator(idxToRemove.size());
+        while (rit.hasPrevious()) {
+            selectedAlignments.remove( rit.previous().intValue() );
+        }
+
+        return new Tuple2<>(selectedAlignments, lowUniquenessMappings);
+    }
+
+    /**
+     * Each alignment has an entry of a tuple2, one for max overlap front, one for max overlap rear,
+     * max overlap front is a tuple2 registering the index and overlap bases count
+     */
+    private static List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> getMaxOverlapPairs(final List<AlignmentInterval> selectedAlignments) {
+
+        final List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> maxOverlapMap =
```

Comment:
I'd make this be a `Map<AlignmentInterval, Tuple2<>>` instead of a `List`.

Reply:
Actually, I think two maps might be better, one for "front" overlap and one for "rear". Then return a tuple2 of the two maps.